src/views/Home.jsx

```diff
@@ -1,14 +1,57 @@
 import './Home.css';
+import { useState, useCallback } from 'react';
+import { getItemData, streamListItems } from '../api';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+
+export function Home({ setListToken }) {
+	const [tokenInput, setTokenInput] = useState('');
+
+	const handleClick = useCallback(() => {
+		const newToken = generateToken();
+		setListToken(newToken);
+	}, [setListToken]);
+
+	const tokenHandler = (e) => {
+		setTokenInput(e.target.value);
+	};
+
+	const handleTokenSubmit = (e) => {
+		e.preventDefault();
+		streamListItems(tokenInput, (snapshot) => {
+			const listData = getItemData(snapshot);
+			if (listData?.length === 0) {
+				alert('this token does not exist');
```

Comment:
For consistency with other views, I'd like to handle this message ourselves rather than use alert. (Just like in `AddItem.jsx`)

Reply:
Also, if we consistently handle alert messages ourselves we could later add a common CSS class to them so they are easily styled the same.